lib/shape.ml

```diff
@@ -4,74 +4,82 @@ type point = { x : int; y : int }
 type line = { a : point; b : point }
 type rectangle = { c : point; length : int; width : int }
 type circle = { c : point; radius : int }
-type ellipse = {c : point; rx : int; ry: int}
+type ellipse = { c : point; rx : int; ry : int }
 type shape = Circle of circle | Rectangle of rectangle | Ellipse of ellipse | Line of line
 type shapes = shape list
 
-let dimensions = ref {x = 500; y = 500}
-let set_dimensions x y = 
-  dimensions := {x;y} 
+let dimensions = ref { x = 500; y = 500 }
+let set_dimensions x y =
+  dimensions := { x; y }
 
-let canvas_mid = { x = (!dimensions.x / 2); y = (!dimensions.y / 2)}
+let canvas_center = { x = 0; y = 0 }
 
 let axes_flag = ref false
-let draw_axes flag = 
+let draw_axes flag =
   axes_flag := flag
 
-let draw_line x1 y1 x2 y2 = 
+let draw_line x1 y1 x2 y2 =
   draw_poly_line [|(x1, y1); (x2, y2)|]
 
+let denormalize x y =
+  { x = x + canvas_center.x; y = -y + canvas_center.y }
```

Comment:
y would get transformed to `y + canvas_mid.y`, correct?
Let me know if I am missing something here.

Reply:
> y would get transformed to `y + canvas_mid.y`, correct? Let me know if I am missing something here.

Yes, you're right